docs/proposals/thin-volumes.md

```diff
@@ -0,0 +1,175 @@
+# Topolvm: Thin Provisioning
+
+## Summary
+Thin provisioning will allow topolvm users to create thin logical volumes using thin-pools.
+
+## Motivation
+- Storage space can be used more effectively. More users can be accommodated for the same amount of storage space when compared to thick provisioning. This significantly reduces upfront hardware cost for the storage admins.
+
+- Faster clones and snapshots.
+
+### Goals
+- As a storage admin, I want to dynamically provision disk space to users on demand based on how much space they need at a given time, so that I can manage unused disk space efficiently.
+
+- As a storage user,  I want to be able to create a large PV, so that I don’t have to worry about extending the PV over and over again.
+
+- As a storage admin, I want to be alerted if the available size of Volume Group with thin logical volumes crosses a certian threshold, so that I can take precautionary measures like extending the volume group.
+
+## Proposal
+- Topolvm is **not** responsible for creating the thin-pools. Storage admin/operator has to create the thin-pools in advance while creating the volume groups.
+- Multiple device classes can be mapped to a single volume group having multiple thin pools. But each device class should map to a single thin pool within that volume group.
+- `lvmd.yaml` file should be updated to hold the mapping between device classes, volume group and thin pools.
+- For device classes mapped to thin pools, the node annotation `capacity.topolvm.cybozu.com/<device-class>` should be updated with respect to the thin pool size and overprovision ratio.
+- Metrics related to thin volume `data` and `metadata` parameters should be exported.
+
+
+
+## Design Details
+- A storage admin has to create the volume groups and the thin-pools on this volume group.
+- The storage admin should update the lvmd.yaml config file on the nodes to map device classes, volume groups and the corresponding thin-pools.
+- Sample `lvmd.yaml` config file:
+``` yaml=
+device-classes:
+  - name: ssd-thin
+    volume-group: myvg1
+    default: true
+    spare-gb: 10
+    type: thin
+    thin-pool:
+        name: pool0
+        overprovision-ratio: 50.0
+  - name: ssd
+    volume-group: myvg1
+    default: true
```

Comment:
There are 2 entries mapping default to true in this example.

Reply:
fixed.